File: Solutions/121- Best Time to Buy and Sell Stock/Solution.java
class Solution {
    public int maxProfit(int[] prices) {

        if (prices.length < 2)
            return 0;

        int answer = 0;
        int min = prices[0];

        for (var price : prices) {
            answer = Math.max(answer, price - min);
            min = Math.min(min, price);
        }
        return answer;
    }
}

File: Solutions/121- Best Time to Buy and Sell Stock/Solution.py
class Solution:
    def maxProfit(self, prices: [int]) -> int:
        if len(prices) < 2:
            return 0

        answer = 0
        minimum = prices[0]

        for price in prices:
            answer = max(answer, price - minimum)
            minimum = min(minimum, price)


        return answer